File: MoodyClues-Journaling-App-working-branch/backend-spring/MoodyClues/src/main/java/com/moodyclues/model/LinkRequest.java
package com.moodyclues.model;

import java.time.LocalDateTime;

import org.hibernate.annotations.UuidGenerator;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;

@Entity
public class LinkRequest {

	@Id
	@GeneratedValue
	@UuidGenerator
	private String id;
	
	@ManyToOne
	private CounsellorUser counsellorUser;
	
	@ManyToOne
	private JournalUser journalUser;
	
	private LocalDateTime requestedAt;
	
	private Status status;
	

    public enum Status {
        PENDING, ACCEPTED, DECLINED
    }
    
    public LinkRequest() {
    	
    }

    
    // GETTERS AND SETTERS
	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public CounsellorUser getCounsellorUser() {
		return counsellorUser;
	}

	public void setCounsellorUser(CounsellorUser counsellorUser) {
		this.counsellorUser = counsellorUser;
	}

	public JournalUser getJournalUser() {
		return journalUser;
	}

	public void setJournalUser(JournalUser journalUser) {
		this.journalUser = journalUser;
	}

	public LocalDateTime getRequestedAt() {
		return requestedAt;
	}

	public void setRequestedAt(LocalDateTime requestedAt) {
		this.requestedAt = requestedAt;
	}

	public Status getStatus() {
		return status;
	}

	public void setStatus(Status status) {
		this.status = status;
	}
    
    
	
}

File: app/src/main/java/com/example/nus/model/LinkRequest.kt
package com.example.nus.model

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

data class LinkRequest(
    @SerializedName("id")
    val id: String,
    
    @SerializedName("counsellorUser")
    val counsellorUser: CounsellorUser?,
    
    @SerializedName("journalUser")
    val journalUser: JournalUser?,
    
    @SerializedName("requestedAt")
    val requestedAt: String, // 使用String来接收ISO格式的日期时间
    
    @SerializedName("status")
    val status: LinkRequestStatus
)

enum class LinkRequestStatus {
    @SerializedName("PENDING")
    PENDING,
    
    @SerializedName("ACCEPTED")
    ACCEPTED,
    
    @SerializedName("DECLINED")
    DECLINED
}

data class CounsellorUser(
    @SerializedName("id")
    val id: String,
    
    @SerializedName("email")
    val email: String,
    
    @SerializedName("firstName")
    val firstName: String?,
    
    @SerializedName("lastName")
    val lastName: String?
)

data class JournalUser(
    @SerializedName("id")
    val id: String,
    
    @SerializedName("email")
    val email: String,
    
    @SerializedName("firstName")
    val firstName: String?,
    
    @SerializedName("lastName")
    val lastName: String?
)

// 创建邀请请求的DTO
data class CounsellorLinkRequestDto(
    @SerializedName("clientEmail")
    val clientEmail: String
)

// 处理邀请决定的DTO
data class JournalLinkRequestDto(
    @SerializedName("approved")
    val approved: Boolean
)

// API响应包装类
data class LinkRequestResponse(
    @SerializedName("success")
    val success: Boolean,
    
    @SerializedName("message")
    val message: String?,
    
    @SerializedName("data")
    val data: LinkRequest?
)
